src/app/utils/useParam.ts

```
@@ -34,7 +34,18 @@ const useParam = <T>(urlParam: UrlParam<T>): [T, SetValueFn<T>] => {
     const newValue = value ? encodeParam(urlParam, value) : null
     const newParams = buildParamQuery(urlParam, newValue)
 
-    history[method]({ ...location, search: newParams.toString() })
+    if (location.pathname !== window.location.pathname) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        "There's a mismatch between window.location and the location from the useLocation hook from react-router. This sometimes happens when calling the 'setValue' inside a non-react event-handler that is not updated. Please be sure the component is updated via react-router",
+        'Tried to update parameter with key: "',
+        urlParam.name,
+        '"',
+      )
+
+      return undefined
+    }
```

Shouldn't this throw instead of show a warning? Seems serious enough to break.

---

won't be resolved when remove the dual setup. I think this is the case: it happens when the values from `useParam` are used in a non-react eventhandler, and the value of eg. `setLocation` is still the same (and holds an 'old' value of `location`, coming from `useLocation`), even after a re-render of the component. Happens a lot with leaflet events. We should make a fix for this in the ARM and react-maps repo.